### PR TITLE
[ADF-4755][CardViewDate&SelectItemComponent] Provide a way to reset date and none option as default.

### PR DESCRIPTION
**Pull request by sivakumar414ram**
**Please check if the PR fulfills these requirements**

> - [x] The commit message follows our [guidelines](https://github.com/Alfresco/alfresco-ng2-components/wiki/Commit-format)
> - [x] Tests for the changes have been added (for bug fixes / features)
> - [x] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x")

> - [ ] Bugfix
> - [x] Feature
> - [ ] Code style update (formatting, local variables)
> - [ ] Refactoring (no functional changes, no api changes)
> - [ ] Build related changes
> - [ ] Documentation
> - [ ] Other... Please describe:

**What is the current behaviour?** (You can also link to an open issue here)

 * There was no way to set date value to empty.
 * There was no way to set select none.

**What is the new behaviour?**

 * Provided a way to set date value to empty.
 * Provided a way to set select none (Added none as the default option and input to toggle default none option).

**Does this PR introduce a breaking change?** (check one with "x")

> - [ ] Yes
> - [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

File: lib/core/card-view/components/card-view-item-dispatcher/card-view-item-dispatcher.component.ts [hunk 1]
```diff
@@ -92,6 +98,8 @@ export class CardViewItemDispatcherComponent implements OnChanges {
         this.componentReference.instance.editable = this.editable;
         this.componentReference.instance.property = this.property;
         this.componentReference.instance.displayEmpty = this.displayEmpty;
+        this.componentReference.instance.displayNoneOption = this.displayNoneOption;
```
Comment (popovicsandras) [inline lib/core/card-view/components/card-view-item-dispatcher/card-view-item-dispatcher.component.ts]: Please add these new fields for testing inside this test suite:
`should update the subcomponent\'s input parameters`

File: lib/core/card-view/components/card-view-selectitem/card-view-selectitem.component.html [hunk 2]
```diff
@@ -4,6 +4,7 @@
     <div *ngIf="isEditable()">
         <mat-form-field>
             <mat-select [(value)]="value" (selectionChange)="onChange($event)" data-automation-class="select-box">
+              <mat-option *ngIf="showNoneOption()">None</mat-option>
```
Comment (mauriziovitale) [inline lib/core/card-view/components/card-view-selectitem/card-view-selectitem.component.html]: None is not translated

Comment (sivakumar414ram) [reply lib/core/card-view/components/card-view-selectitem/card-view-selectitem.component.html]: fixed

File: lib/core/card-view/components/card-view-item-dispatcher/card-view-item-dispatcher.component.ts (hunk 1 above)
Comment (sivakumar414ram) [reply lib/core/card-view/components/card-view-item-dispatcher/card-view-item-dispatcher.component.ts]: done

**Pull request merged**